app/views/games/show.html.erb

```diff
@@ -7,14 +16,39 @@
         $.ajax({
           type: 'PUT',
           url: ui.draggable.data('update-url'),
+          dataType: 'json',
+          context: $(this),
           data: {piece: {row: $(this).parent().index(), column: $(this).index()}}
         }).done(function( data ) {
           $(ui.draggable).detach().css({top: 0, left: 0}).appendTo(drop_space_element);
+          var piece = data.piece
+          var type = data.piece_type
+          if (type === 'Pawn' && is_in_back_row(piece)){
+            pawnId = piece.id;
+            $droppedSpace = $(drop_space_element);
+            $(this).modal('show');
+          }
         }).fail(function() {
           ui.draggable.animate({top: '0px', left: '0px'}, 'slow');
         });
       }
     });
+
+    $('#promote-form').submit(function(event) {
+      var piece_type = $('input[name=promotion_type]:checked', '#promote-form').val()
+      $.ajax({
+        type: 'PUT',
+        url: $(this).attr('action'),
+        dataType: 'json',
+        data: {promotion_type: piece_type, pawn_id: pawnId}
+      }).done(function( data ) {
+        $droppedSpace.html('<span class="piece ui-draggable ui-draggable-handle" data-update-url="/pieces/341" style="position: relative;">'+ data.chess_font_character + '</span>');
+        $('#promote-form').modal('hide');
```

Comment:
@mscoutermarsh This doesn't end up hiding the modal for some reason. I've tried a couple other similar things but nothing is working.